src/main/java/org/opensearch/geospatial/ip2geo/action/RestUpdateDatasourceHandler.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import static org.opensearch.geospatial.shared.URLBuilder.URL_DELIMITER;
+import static org.opensearch.geospatial.shared.URLBuilder.getPluginURLPrefix;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+/**
+ * Rest handler for Ip2Geo datasource update request
+ */
+public class RestUpdateDatasourceHandler extends BaseRestHandler {
+    private static final String ACTION_NAME = "ip2geo_datasource_update";
+
+    @Override
+    public String getName() {
+        return ACTION_NAME;
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        final PutDatasourceRequest putDatasourceRequest = new PutDatasourceRequest(request.param("name"));
+        if (request.hasContentOrSourceParam()) {
```

Comment:
Is it valid if it doesnt have any content?

Reply:
Not valid. Good catch.

Reply:
Added in `validate` method of `UpdateDatasourceRequest` class